src/main/java/com/pinterest/secor/monitoring/MicroMeterMetricCollector.java

```diff
@@ -63,13 +78,22 @@ public void increment(String label, int delta, String topic) {
 
     @Override
     public void metric(String label, double value, String topic) {
-        Metrics.gauge(label, Collections.singletonList(
-            Tag.of("topic", topic)), new AtomicDouble(0)).set(value);
+        gauge(label, value, topic);
     }
 
     @Override
     public void gauge(String label, double value, String topic) {
+        String key = label + "_" + topic;
+        if (!mGaugeCache.containsKey(key) && mGaugeCache.size() >= mConfig.getMicroMeterCacheSize()) {
+            LOG.error("Gauge cache size reached maximum, this may result in inaccurate metrics, "
+                    + "you can increase cache size by changing "
+                    + "\"secor.monitoring.metrics.collector.micrometer.cache.size\" property.");
```

Comment:
Can you put 'return' statement here?  Otherwise Metrics.gauge() result might be un-deterministic